3.15/dual_call_stacks.md

```diff
@@ -0,0 +1,109 @@
+# Dual Call Stacks
+
+Authors: Ken Jin, Pablo Galindo
+
+CPython currently uses a single Python call stack
+(separate from the C call stack). This is in contrast to other runtimes
+like HotSpot and some Wasm runtimes which use dual call stacks.
+In this document, we propose changing CPython's runtime to use dual
+call stacks.
+
+## The Problem
+
+There are two main issues faced with single call stacks:
+1. Traversing the all frames in a thread state is expensive, as
+   the frame layout is roughly a linked list.
```

Comment:
Isn't this resolved by the introduction of `_PyStackChunk`? or am I misreading what is meant by `frames` here? 

`_PyStackChunk` is still a linked list, but is much better than copying all `_PyInterpreterFrame`.

Reply:
Yeah it's a bit subtle, that's what I meant by "roughly" a linked list :).